### ISSUE TITLE
SSO Flow

### ISSUE THREAD

**Issue by rijkvanzanten** [directus/api-next#15]
Currently, the flow is:

1. User clicks SSO button
2. App opens `/:project/auth/sso/:provider`
3. API redirects to provider login page
4. User logs in on provider
5. Provider navigates to `/:project/auth/sso/callback`
**6. API sets cookie and redirects back to app**

We need to figure out how to make step 6 flexible enough to work "natively" in the app, but is also able to return the token normally in case you want to roll your own sso login flow

**Comment by rijkvanzanten** [comment]
We might be able to add a query parameter to step 2 saying if and what the redirect URL should be on login:

```
/:project/auth/sso/:provider ?redirect=/admin
```

The omission of this query parameter indicates that the response should be "regular json"

**Comment by jooola** [comment]
Instead of building our own, we should use the [Laravel passport application](https://laravel.com/docs/6.x/passport).
Maybe SSO can be removed from directus and delegated to laravel. So Directus don't handle auth, let users setup there own auth using passport if sso is required (maybe we should provide some examples), and create some helpers that simplify directus <-> passport bindings.

**Comment by jooola** [comment]
Also, It would be interesting to handle user creation if the user that logged with SSO don't have a user yet.
I know SCIM is supported, but the workflow aren't the same.

**Comment by rijkvanzanten** [comment]
@jooola Absolutely. "Public" user registration (with a default non-admin role) is a highly requested feature. Being able to create these users through SSO should be supported too
